scripts/quests/windurst/Carbuncle_Debacle.lua

```diff
@@ -0,0 +1,186 @@
+                {
+                    [415] = function(player, csid, option, npc)
+                            quest:begin(player)
+                            quest:setVar(player, 'Prog', 1)
+                    end,
+                },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        --Section: Quest Accepted
+        [xi.zone.WINDURST_WALLS] =
+        {
+            ['Koru-Moru'] =
+            {
+                onTrigger = function(player, npc)
+                    local questProgress = quest:getVar(player, 'Prog')
+
+                    if questProgress == 1 then
+                        return quest:event(416)
+                    elseif questProgress == 4 then
+                        return quest:progressEvent(417)
+                    elseif questProgress == 5 then
+                        return quest:progressEvent(418)
+                    elseif questProgress == 7 then
+                        return quest:progressEvent(419)
+                    elseif player:getQuestStatus(xi.quest.log_id.WINDURST, xi.quest.id.windurst.CARBUNCLE_DEBACLE) == QUEST_COMPLETED then
+                        return player:event(420)
+                    end
+                end,
+            },
+                onEventFinish =
+                {
+                    [416] = function(player, csid, option, npc)
+                        quest:setVar(player, 'Prog', 2)
+                    end,
+
+                    [417] = function(player, csid, option, npc)
+                        quest:setVar(player, 'Prog', 5)
+                        npcUtil.giveKeyItem(player, xi.ki.DAZE_BREAKER_CHARM)
+                    end,
+
+                    [419] = function(player, csid, option, npc)
+                        quest:complete(player)
+                        player:addTitle(xi.title.PARAGON_OF_SUMMONER_EXCELLENCE)
+                    end
+                }
+            },
+
+        [xi.zone.MHAURA] =
+        {
+
+            ['Ripapa'] =
+            {
+                onTrigger = function(player, npc)
+                    local questProgress = quest:getVar(player, 'Prog')
+
+                    if questProgress == 2 then
+                        return quest:progressEvent(10022)
+                    elseif questProgress == 3 and not player:hasItem(xi.items.LIGHTNING_PENDULUM) then
+                        return quest:event(10023)
+                    end
+                end,
+            },
+
+                onEventFinish =
+                {
+                    [10022] = function(player, csid, option, npc)
+                        npcUtil.giveItem(player, xi.items.LIGHTNING_PENDULUM)
+                        quest:setVar(player, 'Prog', 3)
+                    end,
+
+                    [10023] = function(player, csid, option, npc)
+                        npcUtil.giveItem(player, xi.items.LIGHTNING_PENDULUM)
+                    end
```

Comment:
Both these events reward lightning pendulum? (Not exactly sure how this quest works)